File: algorithms/1_easy/K/00500_keyboard_row/solution.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        res = []
        flags = [0] * 128

        for ch in "qwertyuiopQWERTYUIOP":
            flags[ord(ch)] = 1

        for ch in "asdfghjklASDFGHJKL":
            flags[ord(ch)] = 2

        for ch in "zxcvbnmZXCVBNM":
            flags[ord(ch)] = 4

        for word in words:
            flag = 0
            for ch in word:
                flag |= flags[ord(ch)]

                if flag != 1 and flag != 2 and flag != 4:
                    break

            if flag == 1 or flag == 2 or flag == 4:
                res.append(word)

        return res

File: algorithms/1_easy/K/00500_keyboard_row/solution.cpp
class Solution {
public:
    vector<string> findWords(vector<string>& words) {
        vector<string> res{};
        array<char, 128> flags{};

        for (const char& ch : "qwertyuiopQWERTYUIOP") {
            flags[static_cast<size_t>(ch)] = 1;
        }

        for (const char& ch : "asdfghjklASDFGHJKL") {
            flags[static_cast<size_t>(ch)] = 2;
        }

        for (const char& ch : "zxcvbnmZXCVBNM") {
            flags[static_cast<size_t>(ch)] = 4;
        }

        for (const string& word : words) {
            char flag{0};

            for (const char& ch : word) {
                flag |= flags[static_cast<size_t>(ch)];

                if (__builtin_popcount(flag) != 1) {
                    break;
                }
            }

            if (__builtin_popcount(flag) == 1) {
                res.push_back(word);
            }
        }

        return res;
    }
};
